plugins/rikki/heroeslounge/models/SlothTeamPivot.php

```diff
@@ -0,0 +1,50 @@
+<?php namespace Rikki\Heroeslounge\Models;
+
+use October\Rain\Database\Pivot;
+use Rikki\Heroeslounge\Models\Sloth;
+use Rikki\Heroeslounge\Models\Team;
+
+/**
+ * Sloth-Team Pivot Model
+ */
+class SlothTeamPivot extends Pivot
+{
+    
+    /*
+     * Disable timestamps by default.
+     * Remove this line if timestamps are defined in the database table.
+     */
+    public $timestamps = false;
+
+    /**
+     * @var string The database table used by the model.
+     */
+    public $table = 'rikki_heroeslounge_sloth_team';
+
+    public function afterSave()
+    {
+        if ($this->isDirty('is_captain')) {
+            $sloth = Sloth::find($this->sloth_id);
+            $noLongerCaptain = false;
+
+            if ($sloth->teams->count() <= 1) {
+                $noLongerCaptain = true;
+            } else {
+                $isCaptain = $sloth->teams->first(function($index, $team) {
+                    return $team->id != $this->team_id && $team->pivot->is_captain;
+                });
+
+                if ($isCaptain != null) {
+                    $noLongerCaptain = true;
+                }
+            }
+
+            if ($this->is_captain) {
+                $sloth->addDiscordCaptainRole();
+            } else if (!$this->is_captain && $noLongerCaptain) {
+                $sloth->removeDiscordCaptainRole();
+            }
```

Comment:
I dislike this part for several reasons. First, it is quite long and for the most time the purpose of this code remains unclear - only in the last few lines is it revealed that it is all for the Discord integration. Maybe a comment or refactoring into a method could help with that.
Second: if ($this->is_captain) holds, $noLongerCaptain is not needed, even though half of the code deals with calculating it. Please only calculate it in the else if block in what is currently line 45 (and make an else out of the else if). While you are at it, rename $noLongerCaptain. It does not really convey the correct meaning, what it actually captures is whether there are other teams that the sloth is a captain of. Also, $isCaptain actually holds a team which is not what the name conveys. At least one of the two variables can be deleted.
Are lines 30/31/32/40 actually needed? From what I can see the else block should be fine even when there is only one team.